### PR TITLE
Improve cache hit rate but ensure we pull the latest software-base

### PR DESCRIPTION
**Pull request by brianhlin**

### CODE REVIEW

File: .github/workflows/build-container.yml [hunk 1]
```diff
@@ -36,6 +36,10 @@ jobs:
       with:
         path: /tmp/.buildx-cache
         key: base-${{ matrix.repo }}-buildx-${{ github.sha }}
+        # allow cache hits from previous runs of the current branch,
+        # parent branch, then upstream branches, in that order
+        restore-keys: |
+          base-${{ matrix.repo }}-buildx-
```
Comment (matyasselmeci) [inline .github/workflows/build-container.yml:42]: Not familiar with restore-keys... do you need a `${{ github.sha }}` at the end? What creates this key?

Comment (brianhlin) [reply .github/workflows/build-container.yml:42]: These docs do a pretty good job of explaining the cache key hit behavior: https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key

tl;dr This `actions/cache` step creates the SHA-specific key and the restore-key `base-${{ matrix.repo }}-buildx-` allows for cache hits that aren't SHA-specific. So excluding the Git SHA allows this step to load the buildx cache from previous GHA runs, speeding up subsequent builds.

Comment (matyasselmeci) [reply .github/workflows/build-container.yml:42]: OK, makes sense.

**Review by matyasselmeci — APPROVED** [review]
